### ISSUE TITLE
Mesos Backend Incorrectly Configures Hostname for Host IPSource

### ISSUE THREAD

**Issue by nemosupremo** [containous/traefik#2545]
### Do you want to request a *feature* or report a *bug*?

Bug.

When you setup the mesos backend, the generated config files results in
the `backend.url` field being set to something like `"http://:47675"` when you
set `IPSource` to host. 

Traefik uses mesosphere/mesos-dns@b47dc4c19f215e98da687b15b4c64e70f629bea5 to 
easily communicate with mesos, however it doesn't fully bootstrap the
configuration with something mesos-dns expects. When Traefik retrieves the 
list of running tasks from Mesos it manually sets `task.SlaveIP` to `slave.Hostname`.

https://github.com/containous/traefik/blob/c66d9de759b4bdebf8e170539eac1bcdd1295894/provider/mesos/config.go#L82-L86

The problem is `task.IP()` expects `task.SlaveIP` to be an actual IPv4 or IPv6, not
a hostname. As a result it tries to parse the hostname as an IP, but fails,
resulting in an empty string as the hostname. 

https://github.com/containous/traefik/blob/c66d9de759b4bdebf8e170539eac1bcdd1295894/provider/mesos/config.go#L207-L209

Traefik should either be using `slave.PID.Host` or a special case in `getHost` to 
properly get the hostname for the task.

### What did you do?

Set up Traefik to use the Mesos Backend

### What did you expect to see?

The host IP in the backend url.

### What did you see instead?

An empty string in the place of an IP address.

### Output of `traefik version`: (_What version of Traefik are you using?_)

containous/traefik@c66d9de759b4bdebf8e170539eac1bcdd1295894

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

```toml
[entryPoints]
  [entryPoints.http]
  address = ":80"

[mesos]
endpoint = "zk://zoo1:2181/mesos"
watch = true
domain = "mesos"
filename = "/mnt/mesos/sandbox/mesos.tmpl"
IPSources = "host"
```

**Comment by nemosupremo** [comment]
I see this issue was reported 7 months ago in #1618, but it was closed. I'm guessing the mesos backend is deprecated or not well supported?

**Comment by timoreimann** [comment]
@nemosupremo my personal impression is that we happen to have a fairly low number of Mesos users, which may an impact on the quality of the provider. In contrary, quite a number of users run Mesos on top of Marathon which has fairly solid support.

Would you have a chance to run on Marathon, or is Mesos a must-have for you?

**Comment by nemosupremo** [comment]
We would prefer to use Mesos as we'd like to use Traefik to route to Spark drivers we run on Mesos. I ended up fixing this specific issue [in my own fork](https://github.com/nemosupremo/traefik/commit/fc1b5ef50f7a2c1d85cd1c3b8adb55f4e5a69d11), as well as adding other functionality to make Mesos template generation a bit easier.

Next, I'm a bit less confident in the Marathon API as I feel Mesosphere puts more effort behind DC/OS and aspects of Marathon have been less than stellar. Currently Marathon 1.5.0+ doesn't even work with Traefik (#2115) because of a backwards incompatible API change.

**Comment by traefiker** [comment]
Closed by #2590.
